### PR TITLE
Implement BlockShotPathTactic

### LINKED ISSUE #460: Implement blockShotPath Tactic

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Implemented the BlockShotPathTactic. It moves the robot into position to block the entire friendly net from the shot (exactly what it sounds like).

### Testing Done

Unit tests for the Tactic added, and I verified it behaves as expected in grsim.

### Resolved Issues

resolves #460 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+Point BlockShotPathTactic::getBlockPosition()
+{
+    if (!this->field)
+    {
+        LOG(WARNING) << "The Field for the " << getName() << " was not initialized"
+                     << std::endl;
+        return this->robot->position();
+    }
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp]: Could we have the field be passed into the constructor? Then we could remove this check.

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp]: Yea that seems reasonable. _Technically_ this is more robust to the field changing, but IRL it should never change while AI is running.

**Comment by MathewMacDougall** [comment]
PR updated

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.cpp]: fixed

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    std::string getName() const override;
+
+    /**
+     * Updates the parameters for this BlockShotPathTactic.
+     *
+     * @param enemy_robot The enemy robot to block from shooting on the friendly net
+     * @param field the Field we are playing on
+     */
+    void updateParams(const Robot& enemy_robot, const Field& field);
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h]: Should this take a field?

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Updates the parameters for this BlockShotPathTactic.
+     *
+     * @param shot_origin The origin of the shot to block
+     * @param field the Field we are playing on
+     */
+    void updateParams(const Point& shot_origin, const Field& field);
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h]: Should this take a field?

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h]: Whoops, not anymore.

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h (hunk 3 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h]: No. fixed

File: src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/block_shot_path_tactic.h]: fixed

**Review by garethellis0 — APPROVED** [review]

**Comment by garethellis0** [comment]
LGTM :+1: :tada:

**Pull request merged**